lib/commands/publish/updateChangedPackages.js

```diff
@@ -19,10 +27,12 @@ function updateDepsObject(deps, changedPackages, versions) {
   }
 }
 
-module.exports = function updateChangedPackages(changedPackages, packagesLoc, versions) {
+module.exports = function updateChangedPackages(changedPackages, packagesLoc, versions, flags) {
   var changedFiles = [];
 
-  changedPackages.forEach(function (pkg) {
+  var packages = flags.canary ? allPackages(packagesLoc) : changedPackages;
```

Comment:
Do we want to do this by default? (Is this related to a current issue?)

If I only modify `pkg1` and other pkgs depend on it (say pkg2), then if we publish something that isn't covered in the range then all those dependencies wouldn't be updated.

Ex: in the case of babel-generator, if we only publish that then I think the users who want to use the newer version of it but only use babel-core wouldn't get the update?